### ISSUE TITLE
weave-npc panic after delete a namespace in k8s cluster

### ISSUE THREAD

**Issue by gobomb** [weaveworks/weave#3836]
## What you expected to happen?

The weave-npc container in k8s clusters could work well and not panic, after I delete a namespace which has some pods and networkpolicies.

## What happened?

When I delete the namespace, a panic (`invalid memory address or nil pointer dereference`) will be observed in weave-npc container in k8s cluster. The panic is at [this line](https://github.com/weaveworks/weave/blob/1ff09fd4d1c52bfa49222b724742f7b4bb80543a/npc/namespace.go#L336)  in the `deletePod` handler.

## How to reproduce it?

It happend randomly.  

1. There are about 20 namespaces in our cluster, there are some pods and networkpolicies in a namespaces.

2. Delete a namespace, sometime we can trigger this panic. 

## Anything else we need to know?

I found why it panic and try to fix it, and I open a PR #3833 . 

When we delete a namespace, the resources like pods belong to this namespace will be deleted first. 

In weave-npc, if the func [deletePod](https://github.com/weaveworks/weave/blob/1ff09fd4d1c52bfa49222b724742f7b4bb80543a/npc/namespace.go#L320)  is called before the func [deleteNamespace](https://github.com/weaveworks/weave/blob/1ff09fd4d1c52bfa49222b724742f7b4bb80543a/npc/namespace.go#L537) , everything is OK. But the sequence is not guaranteed, because these two func are called concurrently.

The func `deleteNamespace`  may be called firstly. The ns pointer will be [set to nil](https://github.com/weaveworks/weave/blob/1ff09fd4d1c52bfa49222b724742f7b4bb80543a/npc/namespace.go#L538) in `deleteNamespace`. This nil pointer will be dereferenced in `deletePod` func (at [this line](https://github.com/weaveworks/weave/blob/1ff09fd4d1c52bfa49222b724742f7b4bb80543a/npc/namespace.go#L336)), which cause panic .

## Versions:

```
$ weave version
weave script 2.6.5
$ docker version
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea838
 Built:             Wed Nov 13 07:22:05 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea838
  Built:            Wed Nov 13 07:28:45 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
$ uname -a
Linux m1.ha.env.lab.io 3.10.0-1062.12.1.el7.x86_64 #1 SMP Tue Feb 4 23:02:59 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.6", GitCommit:"72c30166b2105cd7d3350f2c28a219e6abcd79eb", GitTreeState:"clean", BuildDate:"2020-01-18T23:31:31Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.6", GitCommit:"72c30166b2105cd7d3350f2c28a219e6abcd79eb", GitTreeState:"clean", BuildDate:"2020-01-18T23:23:21Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
Use 'kc' as shortcut of kubectl and 'kd' for 'kubectl describe'
```

## Logs:

```
$ kubectl logs -n kube-system <weave-net-pod> weave-npc
DEBU: 2020/07/28 09:58:33.732653 EVENT UpdatePod {"metadata":{"creationTimestamp":"2020-07-28T09:57:33Z","generateName":"bp-imagee383-6b69c944b4-","labels":{"App":"f2f83717edc2c4009874aa423409ecd08","Blueprint":"bbp-imagee383","Service":"bp-imagee383","app":"bp-imagee383","flowid":"47","pod-template-hash":"6b69c944b4","version":"v1"},"name":"bp-imagee383-6b69c944b4-krjdg","namespace":"f2f83717edc2c4009874aa423409ecd08","resourceVersion":"11866551","selfLink":"/api/v1/namespaces/f2f83717edc2c4009874aa423409ecd08/pods/bp-imagee383-6b69c944b4-krjdg","uid":"fb672013-8ec3-4de4-9505-2d85bc3a95e2"},"spec":{"affinity":{},"containers":[{"image":"harbor.cloud2go.cn/cloudtogo/network-multitool","imagePullPolicy":"Always","name":"bp-imagee383","terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"FallbackToLogsOnError"}],"dnsConfig":{"options":[{"name":"ndots","value":"1"}]},"dnsPolicy":"ClusterFirst","nodeName":"n2.ha.env.lab.io","restartPolicy":"Always","schedulerName":"default-scheduler","securityContext":{},"serviceAccount":"default","serviceAccountName":"default","terminationGracePeriodSeconds":0},"status":{"conditions":[{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:33Z","status":"True","type":"Initialized"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:35Z","status":"True","type":"Ready"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:35Z","status":"True","type":"ContainersReady"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:33Z","status":"True","type":"PodScheduled"}],"hostIP":"10.10.13.47","phase":"Running","podIP":"10.20.0.37","qosClass":"BestEffort","startTime":"2020-07-28T09:57:33Z"}} {"metadata":{"creationTimestamp":"2020-07-28T09:57:33Z","deletionGracePeriodSeconds":0,"deletionTimestamp":"2020-07-28T09:58:33Z","generateName":"bp-imagee383-6b69c944b4-","labels":{"App":"f2f83717edc2c4009874aa423409ecd08","Blueprint":"bbp-imagee383","Service":"bp-imagee383","app":"bp-imagee383","flowid":"47","pod-template-hash":"6b69c944b4","version":"v1"},"name":"bp-imagee383-6b69c944b4-krjdg","namespace":"f2f83717edc2c4009874aa423409ecd08","resourceVersion":"11867377","selfLink":"/api/v1/namespaces/f2f83717edc2c4009874aa423409ecd08/pods/bp-imagee383-6b69c944b4-krjdg","uid":"fb672013-8ec3-4de4-9505-2d85bc3a95e2"},"spec":{"affinity":{},"containers":[{"image":"harbor.cloud2go.cn/cloudtogo/network-multitool","imagePullPolicy":"Always","name":"bp-imagee383","terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"FallbackToLogsOnError"}],"dnsConfig":{"options":[{"name":"ndots","value":"1"}]},"dnsPolicy":"ClusterFirst","nodeName":"n2.ha.env.lab.io","restartPolicy":"Always","schedulerName":"default-scheduler","securityContext":{},"serviceAccount":"default","serviceAccountName":"default","terminationGracePeriodSeconds":0},"status":{"conditions":[{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:33Z","status":"True","type":"Initialized"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:35Z","status":"True","type":"Ready"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:35Z","status":"True","type":"ContainersReady"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:33Z","status":"True","type":"PodScheduled"}],"hostIP":"10.10.13.47","phase":"Running","podIP":"10.20.0.37","qosClass":"BestEffort","startTime":"2020-07-28T09:57:33Z"}}
DEBU: 2020/07/28 09:58:33.740482 EVENT DeletePod {"metadata":{"creationTimestamp":"2020-07-28T09:57:33Z","deletionGracePeriodSeconds":0,"deletionTimestamp":"2020-07-28T09:58:33Z","generateName":"bp-imagee383-6b69c944b4-","labels":{"App":"f2f83717edc2c4009874aa423409ecd08","Blueprint":"bbp-imagee383","Service":"bp-imagee383","app":"bp-imagee383","flowid":"47","pod-template-hash":"6b69c944b4","version":"v1"},"name":"bp-imagee383-6b69c944b4-krjdg","namespace":"f2f83717edc2c4009874aa423409ecd08","resourceVersion":"11867378","selfLink":"/api/v1/namespaces/f2f83717edc2c4009874aa423409ecd08/pods/bp-imagee383-6b69c944b4-krjdg","uid":"fb672013-8ec3-4de4-9505-2d85bc3a95e2"},"spec":{"affinity":{},"containers":[{"image":"harbor.cloud2go.cn/cloudtogo/network-multitool","imagePullPolicy":"Always","name":"bp-imagee383","terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"FallbackToLogsOnError"}],"dnsConfig":{"options":[{"name":"ndots","value":"1"}]},"dnsPolicy":"ClusterFirst","nodeName":"n2.ha.env.lab.io","restartPolicy":"Always","schedulerName":"default-scheduler","securityContext":{},"serviceAccount":"default","serviceAccountName":"default","terminationGracePeriodSeconds":0},"status":{"conditions":[{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:33Z","status":"True","type":"Initialized"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:35Z","status":"True","type":"Ready"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:35Z","status":"True","type":"ContainersReady"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:33Z","status":"True","type":"PodScheduled"}],"hostIP":"10.10.13.47","phase":"Running","podIP":"10.20.0.37","qosClass":"BestEffort","startTime":"2020-07-28T09:57:33Z"}}
INFO: 2020/07/28 09:58:33.740543 deleting entry 10.20.0.37 from weave-hkFEnqOYZ~#jp*[N$?})[#4t3 of fb672013-8ec3-4de4-9505-2d85bc3a95e2
INFO: 2020/07/28 09:58:33.740565 deleted entry 10.20.0.37 from weave-hkFEnqOYZ~#jp*[N$?})[#4t3 of fb672013-8ec3-4de4-9505-2d85bc3a95e2
INFO: 2020/07/28 09:58:33.761192 deleting entry 10.20.0.37 from weave-e$G8cjWOPOD!AV1BHcQc@]W7T of fb672013-8ec3-4de4-9505-2d85bc3a95e2
INFO: 2020/07/28 09:58:33.761224 deleted entry 10.20.0.37 from weave-e$G8cjWOPOD!AV1BHcQc@]W7T of fb672013-8ec3-4de4-9505-2d85bc3a95e2
INFO: 2020/07/28 09:58:33.762157 deleting entry 10.20.0.37 from weave-6t$=mTtRDZ$mqByw*5EuD[!%6 of fb672013-8ec3-4de4-9505-2d85bc3a95e2
INFO: 2020/07/28 09:58:33.762181 deleted entry 10.20.0.37 from weave-6t$=mTtRDZ$mqByw*5EuD[!%6 of fb672013-8ec3-4de4-9505-2d85bc3a95e2
INFO: 2020/07/28 09:58:33.763892 EVENT UpdateNamespace {"metadata":{"creationTimestamp":"2020-07-28T09:57:29Z","deletionTimestamp":"2020-07-28T09:58:28Z","labels":{"flowid":"47","namespace":"f2f83717edc2c4009874aa423409ecd08"},"name":"f2f83717edc2c4009874aa423409ecd08","resourceVersion":"11867260","selfLink":"/api/v1/namespaces/f2f83717edc2c4009874aa423409ecd08","uid":"5349e4bf-baf0-49a4-a1bf-db7d65c7b048"},"spec":{"finalizers":["kubernetes"]},"status":{"phase":"Terminating"}} {"metadata":{"creationTimestamp":"2020-07-28T09:57:29Z","deletionTimestamp":"2020-07-28T09:58:28Z","labels":{"flowid":"47","namespace":"f2f83717edc2c4009874aa423409ecd08"},"name":"f2f83717edc2c4009874aa423409ecd08","resourceVersion":"11867383","selfLink":"/api/v1/namespaces/f2f83717edc2c4009874aa423409ecd08","uid":"5349e4bf-baf0-49a4-a1bf-db7d65c7b048"},"spec":{"finalizers":["kubernetes"]},"status":{"phase":"Terminating"}}
INFO: 2020/07/28 09:58:33.764160 EVENT DeleteNamespace {"metadata":{"creationTimestamp":"2020-07-28T09:57:29Z","deletionTimestamp":"2020-07-28T09:58:28Z","labels":{"flowid":"47","namespace":"f2f83717edc2c4009874aa423409ecd08"},"name":"f2f83717edc2c4009874aa423409ecd08","resourceVersion":"11867384","selfLink":"/api/v1/namespaces/f2f83717edc2c4009874aa423409ecd08","uid":"5349e4bf-baf0-49a4-a1bf-db7d65c7b048"},"spec":{"finalizers":["kubernetes"]},"status":{"phase":"Terminating"}}
DEBU: 2020/07/28 09:58:33.764191 removing rule for DefaultAllow in namespace: f2f83717edc2c4009874aa423409ecd08, chain: WEAVE-NPC-DEFAULT, [-m set --match-set weave-hkFEnqOYZ~#jp*[N$?})[#4t3 dst -j ACCEPT -m comment --comment DefaultAllow ingress isolation for namespace: f2f83717edc2c4009874aa423409ecd08]
DEBU: 2020/07/28 09:58:33.784954 removing rule for DefaultAllow in namespace: f2f83717edc2c4009874aa423409ecd08, chain: WEAVE-NPC-EGRESS-DEFAULT, [-m set --match-set weave-e$G8cjWOPOD!AV1BHcQc@]W7T src -j WEAVE-NPC-EGRESS-ACCEPT -m comment --comment DefaultAllow egress isolation for namespace: f2f83717edc2c4009874aa423409ecd08]
DEBU: 2020/07/28 09:58:33.788475 removing rule for DefaultAllow in namespace: f2f83717edc2c4009874aa423409ecd08, chain: WEAVE-NPC-EGRESS-DEFAULT, [-m set --match-set weave-e$G8cjWOPOD!AV1BHcQc@]W7T src -j RETURN -m comment --comment DefaultAllow egress isolation for namespace: f2f83717edc2c4009874aa423409ecd08]
DEBU: 2020/07/28 09:58:33.806033 EVENT UpdatePod {"metadata":{"creationTimestamp":"2020-07-28T09:57:35Z","generateName":"bp-s1-569648b84-","labels":{"App":"f2f83717edc2c4009874aa423409ecd08","Blueprint":"bbp-s1","Service":"bp-s1","app":"bp-s1","flowid":"47","pod-template-hash":"569648b84","version":"v1"},"name":"bp-s1-569648b84-jltfv","namespace":"f2f83717edc2c4009874aa423409ecd08","resourceVersion":"11866586","selfLink":"/api/v1/namespaces/f2f83717edc2c4009874aa423409ecd08/pods/bp-s1-569648b84-jltfv","uid":"f990ffe2-227f-4b3e-a1e5-e0f7d59e7c49"},"spec":{"affinity":{},"containers":[{"image":"harbor.cloud2go.cn/cloudtogo/network-multitool","imagePullPolicy":"Always","name":"bp-s1","ports":[{"containerPort":80,"name":"p80","protocol":"TCP"}],"readinessProbe":{"failureThreshold":3,"initialDelaySeconds":1,"periodSeconds":5,"successThreshold":1,"tcpSocket":{"port":80},"timeoutSeconds":1},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"FallbackToLogsOnError"}],"dnsConfig":{"options":[{"name":"ndots","value":"1"}]},"dnsPolicy":"ClusterFirst","nodeName":"n1.ha.env.lab.io","restartPolicy":"Always","schedulerName":"default-scheduler","securityContext":{},"serviceAccount":"default","serviceAccountName":"default","terminationGracePeriodSeconds":0},"status":{"conditions":[{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:35Z","status":"True","type":"Initialized"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:38Z","status":"True","type":"Ready"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:38Z","status":"True","type":"ContainersReady"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:35Z","status":"True","type":"PodScheduled"}],"hostIP":"10.10.13.46","phase":"Running","podIP":"10.20.128.40","qosClass":"BestEffort","startTime":"2020-07-28T09:57:35Z"}} {"metadata":{"creationTimestamp":"2020-07-28T09:57:35Z","deletionGracePeriodSeconds":0,"deletionTimestamp":"2020-07-28T09:58:33Z","generateName":"bp-s1-569648b84-","labels":{"App":"f2f83717edc2c4009874aa423409ecd08","Blueprint":"bbp-s1","Service":"bp-s1","app":"bp-s1","flowid":"47","pod-template-hash":"569648b84","version":"v1"},"name":"bp-s1-569648b84-jltfv","namespace":"f2f83717edc2c4009874aa423409ecd08","resourceVersion":"11867379","selfLink":"/api/v1/namespaces/f2f83717edc2c4009874aa423409ecd08/pods/bp-s1-569648b84-jltfv","uid":"f990ffe2-227f-4b3e-a1e5-e0f7d59e7c49"},"spec":{"affinity":{},"containers":[{"image":"harbor.cloud2go.cn/cloudtogo/network-multitool","imagePullPolicy":"Always","name":"bp-s1","ports":[{"containerPort":80,"name":"p80","protocol":"TCP"}],"readinessProbe":{"failureThreshold":3,"initialDelaySeconds":1,"periodSeconds":5,"successThreshold":1,"tcpSocket":{"port":80},"timeoutSeconds":1},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"FallbackToLogsOnError"}],"dnsConfig":{"options":[{"name":"ndots","value":"1"}]},"dnsPolicy":"ClusterFirst","nodeName":"n1.ha.env.lab.io","restartPolicy":"Always","schedulerName":"default-scheduler","securityContext":{},"serviceAccount":"default","serviceAccountName":"default","terminationGracePeriodSeconds":0},"status":{"conditions":[{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:35Z","status":"True","type":"Initialized"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:38Z","status":"True","type":"Ready"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:38Z","status":"True","type":"ContainersReady"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:35Z","status":"True","type":"PodScheduled"}],"hostIP":"10.10.13.46","phase":"Running","podIP":"10.20.128.40","qosClass":"BestEffort","startTime":"2020-07-28T09:57:35Z"}}
DEBU: 2020/07/28 09:58:33.806374 EVENT DeletePod {"metadata":{"creationTimestamp":"2020-07-28T09:57:35Z","deletionGracePeriodSeconds":0,"deletionTimestamp":"2020-07-28T09:58:33Z","generateName":"bp-s1-569648b84-","labels":{"App":"f2f83717edc2c4009874aa423409ecd08","Blueprint":"bbp-s1","Service":"bp-s1","app":"bp-s1","flowid":"47","pod-template-hash":"569648b84","version":"v1"},"name":"bp-s1-569648b84-jltfv","namespace":"f2f83717edc2c4009874aa423409ecd08","resourceVersion":"11867380","selfLink":"/api/v1/namespaces/f2f83717edc2c4009874aa423409ecd08/pods/bp-s1-569648b84-jltfv","uid":"f990ffe2-227f-4b3e-a1e5-e0f7d59e7c49"},"spec":{"affinity":{},"containers":[{"image":"harbor.cloud2go.cn/cloudtogo/network-multitool","imagePullPolicy":"Always","name":"bp-s1","ports":[{"containerPort":80,"name":"p80","protocol":"TCP"}],"readinessProbe":{"failureThreshold":3,"initialDelaySeconds":1,"periodSeconds":5,"successThreshold":1,"tcpSocket":{"port":80},"timeoutSeconds":1},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"FallbackToLogsOnError"}],"dnsConfig":{"options":[{"name":"ndots","value":"1"}]},"dnsPolicy":"ClusterFirst","nodeName":"n1.ha.env.lab.io","restartPolicy":"Always","schedulerName":"default-scheduler","securityContext":{},"serviceAccount":"default","serviceAccountName":"default","terminationGracePeriodSeconds":0},"status":{"conditions":[{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:35Z","status":"True","type":"Initialized"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:38Z","status":"True","type":"Ready"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:38Z","status":"True","type":"ContainersReady"},{"lastProbeTime":null,"lastTransitionTime":"2020-07-28T09:57:35Z","status":"True","type":"PodScheduled"}],"hostIP":"10.10.13.46","phase":"Running","podIP":"10.20.128.40","qosClass":"BestEffort","startTime":"2020-07-28T09:57:35Z"}}
INFO: 2020/07/28 09:58:33.806394 deleting entry 10.20.128.40 from weave-hkFEnqOYZ~#jp*[N$?})[#4t3 of f990ffe2-227f-4b3e-a1e5-e0f7d59e7c49
INFO: 2020/07/28 09:58:33.806403 deleting entry 10.20.128.40 from weave-e$G8cjWOPOD!AV1BHcQc@]W7T of f990ffe2-227f-4b3e-a1e5-e0f7d59e7c49
INFO: 2020/07/28 09:58:33.806410 deleting entry 10.20.128.40 from weave-6t$=mTtRDZ$mqByw*5EuD[!%6 of f990ffe2-227f-4b3e-a1e5-e0f7d59e7c49
INFO: 2020/07/28 09:58:33.806418 deleted entry 10.20.128.40 from weave-6t$=mTtRDZ$mqByw*5EuD[!%6 of f990ffe2-227f-4b3e-a1e5-e0f7d59e7c49
ERROR: logging before flag.Parse: E0728 09:58:33.807381    7656 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:679
/usr/local/go/src/runtime/panic.go:199
/usr/local/go/src/runtime/signal_unix.go:394
/go/src/github.com/weaveworks/weave/npc/namespace.go:336
/go/src/github.com/weaveworks/weave/npc/controller.go:144
/go/src/github.com/weaveworks/weave/npc/controller.go:106
/go/src/github.com/weaveworks/weave/npc/controller.go:143
/go/src/github.com/weaveworks/weave/prog/weave-npc/main.go:294
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:209
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:320
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:444
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:150
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/weaveworks/weave/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:124
/usr/local/go/src/runtime/asm_amd64.s:1357
```

**Comment by bboreham** [comment]
Thanks for the very detailed report.
I will comment on the PR.

**Comment by bboreham** [comment]
Closed by #3833
